spanner/client.go

```diff
@@ -533,7 +533,8 @@ func (c *Client) rwTransaction(ctx context.Context, f func(context.Context, *Rea
 		}
 		if t.shouldExplicitBegin(attempt) {
 			if err = t.begin(ctx); err != nil {
-				return spannerErrorf(codes.Internal, "error while BeginTransaction during retrying a ReadWrite transaction: %v", err)
+				trace.TracePrintf(ctx, nil, "Error while BeginTransaction during retrying a ReadWrite transaction: %v", ToSpannerError(err))
+				return ToSpannerError(err)
```

Comment:
Could we have a test that shows that this actually fixes the problem? I'm a little surprised that the sample code given in the original issue gets into this branch of the code, as it indicates that the transaction should use an explicit `BeginTransaction` call. It's not clear to me why the example would do that, as it just uses a normal `SELECT` statement as the first statement in the transaction. That would then normally mean that the `BeginTransaction` is inlined with the execution of that statement.

Reply:
Added tests, this happend because we do explicit begin transaction if first statement returns error, in this case call the iter.Next() returned cancelled and while doing explicit Begin we returned customer internal error. https://github.com/googleapis/google-cloud-go/blob/main/spanner/read.go#L143